File: app/src/main/java/com/example/pam_project/lists/categories/components/CategoryInformation.java
package com.example.pam_project.lists.categories.components;

import com.example.pam_project.utils.AppColor;

import java.io.Serializable;

public class CategoryInformation implements Serializable, Comparable<CategoryInformation> {
    private final String title;
    private final AppColor color;
    private long id;

    public CategoryInformation(final String title, final AppColor color) {
        this.title = title;
        this.color = color;
    }

    public CategoryInformation(final String title, final String color) {
        this(title, AppColor.fromName(color));
    }

    public CategoryInformation(final long id, final String title, final AppColor color) {
        this(title, color);
        this.id = id;
    }

    public CategoryInformation(final long id, final String title, final String color) {
        this(id, title, AppColor.fromName(color));
    }

    public long getId() {
        return id;
    }

    public String getTitle() {
        return title;
    }

    public AppColor getColor() {
        return color;
    }

    @Override
    public int compareTo(CategoryInformation categoryInformation) {
        return title.compareTo(categoryInformation.getTitle());
    }
}

File: app/src/main/java/com/example/pam_project/features/categories/list/CategoryInformation.kt
package com.example.pam_project.features.categories.list

import com.example.pam_project.utils.constants.AppColor
import java.io.Serializable

open class CategoryInformation(title: String, color: AppColor?) : Serializable, Comparable<CategoryInformation> {
    var title: String = title
        private set
    var color: AppColor? = color
        private set
    var id: Long = 0
        private set

    constructor(title: String, color: String?) : this(title, AppColor.fromName(color))
    constructor(id: Long, title: String, color: AppColor?) : this(title, color) {
        this.id = id
    }

    constructor(id: Long, title: String, color: String?) : this(id, title, AppColor.fromName(color))

    override fun compareTo(other: CategoryInformation): Int {
        return title.compareTo(other.title)
    }
}